src/deploy/functions/validate.ts

```diff
@@ -85,3 +79,26 @@ export function packageJsonIsValid(
     throw new FirebaseError(msg);
   }
 }
+
+function _functionsSourceIsPresent(data: any, sourceDir: string, projectDir: string): void {
+  const indexJsFile = path.join(sourceDir, data.main || "index.js");
+  if (!fsutils.fileExistsSync(indexJsFile)) {
+    const msg = `${path.relative(
+      projectDir,
+      indexJsFile
+    )} does not exist, can't deploy Firebase Functions`;
+    throw new FirebaseError(msg);
+  }
+}
+
+function _enginesFieldIsPresent(data: any, sourceDirName: string): void {
+  if (!data.engines || !data.engines.node) {
+    const msg =
+      `Engines field is required but was not found in ${sourceDirName}${path.sep}package.json.\n` +
+      `Add the following lines to your package.json to fix this: \n
```

Comment:
I think "To fix this, add the following lines to your package.json" sound better but not a big deal either way.

Reply:
Changed.